src/callbacks/wandb_callbacks.py

```diff
@@ -196,3 +196,40 @@ def on_validation_epoch_end(self, trainer, pl_module):
 
             self.preds.clear()
             self.targets.clear()
+            
+
+class ImagePredictionLogger(Callback):
+    """Logs a validation batch and their predictions to wandb.
+       Example adapted from:
+       https://wandb.ai/wandb/wandb-lightning/reports/Image-Classification-using-PyTorch-Lightning--VmlldzoyODk1NzY
+    """
+
+    def __init__(self, num_samples):
+        super().__init__()
+        self.num_samples = num_samples
+
+    def on_sanity_check_start(self, trainer, pl_module):
+        self.ready = False
+
+    def on_sanity_check_end(self, trainer, pl_module):
+        """Start executing this callback only after all validation sanity checks end."""
+        self.ready = True
+
+    def on_validation_epoch_end(self, trainer, pl_module):
+        # get a validation batch from the calidation dat loader
+        val_samples = next(iter(trainer.datamodule.val_dataloader()))
+        val_imgs, val_labels = val_samples[0], val_samples[1]
+
+        # run the batch through the network
+        val_imgs = val_imgs.to(device=pl_module.device)
+        logits = pl_module(val_imgs)
+        preds = torch.argmax(logits, axis=-1)
+
+        # log the images as wandb Image
+        trainer.logger.experiment[0].log({
```

Comment:
Please use `get_wandb_logger()` method from top of the file when accessing logger. This way the callback will also work if you use many different loggers at once, and a proper exception will be thrown when wandb logger is not available.

Reply:
https://github.com/ashleve/lightning-hydra-template/blob/74eb7c7e57f16607d6b8ae62a627b343248e41a7/src/callbacks/wandb_callbacks.py#L15-L26